### PR TITLE
Adding docs for App Configuration

### PR DESCRIPTION
**Pull request by richardpark-msft**

### CODE REVIEW

File: sdk/appconfiguration/app-configuration/src/appConfigCredential.ts [hunk 1]
```diff
@@ -10,6 +10,12 @@ export class AppConfigCredential implements ServiceClientCredentials {
     this.secret = secret;
   }
 
+  /**
+   * Signs a request with the values provided in the credential and secret parameter.
+   *
+   * @param {WebResource} webResource The WebResource to be signed.
+   * @returns {Promise<WebResource>} The signed request object.
```
Comment (chradek) [inline sdk/appconfiguration/app-configuration/src/appConfigCredential.ts:21]: I think we might want to mark this method with the `@internal` and `@ignore` tags. It probably has to be a public method since it's used by core-http at some point, but we wouldn't expect customers to call it. Those flags will prevent the method from showing up in our documentation.

Comment (chradek) [reply sdk/appconfiguration/app-configuration/src/appConfigCredential.ts:21]: As discussed, actually the whole class should be marked internal since it isn't exported from the package, only from this file for internal use.

File: sdk/appconfiguration/app-configuration/src/index.ts [hunk 2]
```diff
@@ -47,30 +47,51 @@ const deserializationContentTypes = {
...
-export type AddConfigurationSettingConfig = Pick<ModelConfigurationSetting, Exclude<keyof ModelConfigurationSetting, "key">>;
-export type AddConfigurationSettingOptions = ModelConfigurationClientCreateOrUpdateConfigurationSettingOptionalParams;
-export type AddConfigurationSettingsResponse = ModelCreateOrUpdateConfigurationSettingResponse;
+type CreateOrUpdateOptions = ModelConfigurationClientCreateOrUpdateConfigurationSettingOptionalParams;
+type CreateOrUpdateConfigurationSetting = Pick<ModelConfigurationSetting, Exclude<keyof ModelConfigurationSetting, "key">>;
+type CreateOrUpdateResponse = ModelCreateOrUpdateConfigurationSettingResponse;
+
+export type AddConfigurationSettingConfig = CreateOrUpdateConfigurationSetting;
+export type AddConfigurationSettingOptions = CreateOrUpdateOptions;
+export type AddConfigurationSettingsResponse = CreateOrUpdateResponse;
```
Comment (richardpark-msft) [inline sdk/appconfiguration/app-configuration/src/index.ts]: I also just realized how silly this change is - it's a rename of what is _already_ being renamed above. Will fix.

File: sdk/appconfiguration/app-configuration/src/appConfigCredential.ts (hunk 1 above)
Comment (richardpark-msft) [reply sdk/appconfiguration/app-configuration/src/appConfigCredential.ts:21]: Done.

File: sdk/appconfiguration/app-configuration/src/index.ts [hunk 3]
```diff
@@ -47,30 +47,50 @@ const deserializationContentTypes = {
...
   private client: ConfigurationClient;
 
+  /**
+   * Initializes a new instance of the AppConfigurationClient class.
+   * @param connectionString Connection string needed for a client to connect to Azure.
+   *//**
+* Initializes a new instance of the AppConfigurationClient class.
+* @param uri The base URI for the Azure service
+* @param credential The credentials to use for authentication.
+*/
```
Comment (ramya-rao-a) [inline sdk/appconfiguration/app-configuration/src/index.ts]: Fro constructor overloads, you can add the jsdocs above each variation.
Look at [constructors for EventHubsClient](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/src/eventHubClient.ts#L232) for an example

File: sdk/appconfiguration/app-configuration/src/index.ts [hunk 4]
```diff
@@ -91,6 +111,21 @@ export class AppConfigurationClient {
...
+  /**
+   * Add a setting into the Azure App Configuration service, failing if it 
+   * already exists.
+   * 
+   * Example usage:
+   * ```ts
+   * const key = "MyKey";
+   * const label = "MyLabel";
+   * const value = "MyValue";
+   * const result = await client.addConfigurationSetting(key, { label, value });
```
Comment (ramya-rao-a) [inline sdk/appconfiguration/app-configuration/src/index.ts]: I would collapse the 4 line example into a single line by inlining the arguments as the emphasis is on `addConfigurationSetting`.
Also, in editors this would occupy less space and would require no scrolling

File: sdk/appconfiguration/app-configuration/src/index.ts [hunk 5]
```diff
@@ -91,6 +111,21 @@ export class AppConfigurationClient {
...
+   * ```
+   * @param key The name of the key.
+   * @param configSettings A configuration value.
+   * @param options Optional parameters for the value being added.
```
Comment (ramya-rao-a) [inline sdk/appconfiguration/app-configuration/src/index.ts]: The optional parameters are not for the value being added.
They are for the process of adding the value. Maybe reword this?

File: sdk/appconfiguration/app-configuration/src/index.ts [hunk 6]
```diff
@@ -118,18 +164,69 @@ export class AppConfigurationClient {
...
     return this.client.getConfigurationSetting(key, options);
   }
 
+  /**
+   * Lists settings from the Azure App Configuration service, optionally filtered by label, 
+   * accept date time or name.
+   * 
+   * Example code:
+   * ```ts
+   * // find all settings with label "MyLabel"
```
Comment (ramya-rao-a) [inline sdk/appconfiguration/app-configuration/src/index.ts]: Since the description above the example code talks about how the list method works, this comment inside the sample code doesnt add extra value and so is not required.

File: sdk/appconfiguration/app-configuration/src/index.ts [hunk 7]
```diff
@@ -118,18 +164,69 @@ export class AppConfigurationClient {
...
+   * Example code:
+   * ```ts
+   * // create a new setting
+   * await client.addConfigurationSetting(key, { value: "foo1" });
+   * 
+   * // update it with a new value
+   * await client.setConfigurationSetting(key, { value: "foo2" });
+   * 
+   * // retrieves all revisions matching that label
+   * const result = await client.listRevisions({ key: [key] });
```
Comment (ramya-rao-a) [inline sdk/appconfiguration/app-configuration/src/index.ts]: The samples are mainly for the usage of the method in question. I wonder if prior setup is needed here and that the stand alone files would be a better place for this.
@bterlson thoughts?

File: sdk/appconfiguration/app-configuration/src/index.ts (hunk 3 above)
Comment (richardpark-msft) [reply sdk/appconfiguration/app-configuration/src/index.ts]: Done. This syntax also works (verified by generating with typedoc) but it's pretty gross comparatively.

File: sdk/appconfiguration/app-configuration/src/index.ts (hunk 4 above)
Comment (richardpark-msft) [reply sdk/appconfiguration/app-configuration/src/index.ts]: Yes, agree. The keys used to be things like 'foo' and such and they wren't descriptive enough. Not a problem now.

File: sdk/appconfiguration/app-configuration/src/index.ts (hunk 5 above)
Comment (richardpark-msft) [reply sdk/appconfiguration/app-configuration/src/index.ts]: I'll reword it. 

The reason I had that is, for whatever reason, 'label' is _also_ in the optional arguments (as well as in the main 'configSettings' argument as well).

File: sdk/appconfiguration/app-configuration/src/index.ts (hunk 7 above)
Comment (richardpark-msft) [reply sdk/appconfiguration/app-configuration/src/index.ts]: I removed it for now and tried to make it a bit more obvious by changin the 'result' return value to have a name (revisionsForMyKey).
